macros/blueprint-proc-macro/src/job.rs

```diff
@@ -255,7 +255,9 @@ pub(crate) fn generate_event_workflow_tokenstream(
                     // If is_raw, assume the actual context is the second param
                     quote! { ctx. #field_in_self .clone() }
                 })
-                .ok_or_else(|| syn::Error::new(Span::call_site(), "Must specify a context"))?;
+                .ok_or_else(|| {
+                    syn::Error::new_spanned(quote! { (#(#params),*) }, "Must specify a context")
+                })?;
```

Comment:
This not what I meant when I mentioned using `params` instead of `call_site`. params may be already carrying the original span where it has been defined, if not (i.e we created these params somewhere without spans) we should pass to this function `get_event_handler_args` a span, from the original declaration of the job macro.

If something like this not possible, or hard to do, using call_site here is _okay_

Reply:
Yeah, `params` does not work. I have to wrap it as seen above.

Reply:
This won't give you the correct location of the error, it will mark the whole job as an error.